src/webportal/src/app/user/fabric/user-profile.jsx

```diff
@@ -139,6 +146,39 @@ const UserProfile = () => {
     });
   });
 
+  // click `add public ssh keys button` -> open dialog
+  const onAddPublicKeys = useCallback(async sshPublicKeys => {
+    setSSHProcessing(true);
+    let updatedSSHPublickeys = [];
+    if (userInfo.extension.sshKeys) {
+      updatedSSHPublickeys = cloneDeep(userInfo.extension.sshKeys);
+    }
+    updatedSSHPublickeys.push({
+      title: sshPublicKeys.title,
+      value: sshPublicKeys.value,
+      time: sshPublicKeys.time,
+    });
+    await updateUserRequest(userInfo.username, updatedSSHPublickeys);
+    const updatedUserInfo = await getUserRequest(userInfo.username);
+    setUserInfo(updatedUserInfo);
+    setSSHProcessing(false);
+  });
+
+  const onDeleteSSHkeys = useCallback(async sshPublicKeys => {
+    let updatedSSHPublickeys = [];
+    if (userInfo.extension.sshKeys) {
+      updatedSSHPublickeys = cloneDeep(userInfo.extension.sshKeys);
+    }
+    updatedSSHPublickeys = updatedSSHPublickeys.filter(
+      item =>
+        item.title !== sshPublicKeys.title &&
```

Comment:
Please only filter with `item.title !== sshPublicKeys.title`.

Title should be unique and delete means delete an ssh key with a certain title.

Reply:
ok